### PR TITLE
feat(aws-autoscaling): add flag and aspect to require imdsv2

### PR DESCRIPTION
**Pull request by jericht**
Partially fixes: https://github.com/aws/aws-cdk/issues/5137
Related PR: https://github.com/aws/aws-cdk/pull/16051

**Note:** I have some concerns about duplicated code between this and the above linked PR. Please see that PR for more details.

### Changes
Adds an aspect that can enable/disable IMDSv1 on AutoScalingGroups

### Testing
Added unit tests

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by jericht** [comment]
@comcalvi I just rebased this and it is ready for review. Will you be able to take a look at it?

**Comment by comcalvi** [comment]
@jericht Thanks for the PR! I'll definitely take a look at this.

File: packages/@aws-cdk/aws-autoscaling/test/aspects/imds-aspect.test.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      // GIVEN
+      const asg = new AutoScalingGroup(stack, 'AutoScalingGroup', {
+        vpc,
+        instanceType: new ec2.InstanceType('t2.micro'),
+        machineImage: ec2.MachineImage.latestAmazonLinux(),
+      });
+      const aspect = new AutoScalingGroupImdsAspect({ enableImdsV1 });
+
+      // WHEN
+      aspect.visit(asg);
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-autoscaling/test/aspects/imds-aspect.test.ts:92]: Similarly to above:

```suggestion
     cdk.Aspects.of(stack).add(aspect);
```

**Review by comcalvi — CHANGES_REQUESTED** [review]
Overall, looks great! Thank you for your patience in this review. I have some comments below; in addition, we should add a property to autoscaling groups to enable / disable imdsv1.

**Comment by jericht** [comment]
@comcalvi I also wanted to get an opinion on my comments about the duplicated code between this PR and the [ec2 PR](https://github.com/aws/aws-cdk/pull/16051). Do you think what we have right now is fine or should we reorganize these files?

File: packages/@aws-cdk/aws-autoscaling/README.md [hunk 2]
```diff
@@ -378,6 +378,23 @@ new autoscaling.AutoScalingGroup(stack, 'ASG', {
...
+The following example demonstrates how to use the `AutoScalingGroupImdsAspect` to disable IMDSv1 (thus enforcing IMDSv2) for all AutoScalingGroups in a stack:
+
+```ts
+const aspect = new autoscaling.AutoScalingGroupImdsAspect({
+  enableImdsV1: false,
+});
+
+Aspects.of(stack).add(aspect);
+```
+
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-autoscaling/README.md:397]: This should also show an example of creating an `AutoScalingGroup` with the `disableImdsv1` prop.

**Review by comcalvi — CHANGES_REQUESTED** [review]
We're almost there! It would be helpful if you could elaborate on the "suppress warnings" functionality; I don't fully understand what that's doing.

**Pull request merged**